src/tests/homework.spec.js

```diff
@@ -0,0 +1,68 @@
+import { test } from "@playwright/test";
+import { EnrollmentPage } from "./homework.enrollment.page";
+import { password, username, userFullName } from "../fixtures/fixtures";
+
+test.describe("Registration form", async () => {
+
+    test.beforeEach(async ({ page }) => {
+        const enrollmentPage = new EnrollmentPage(page);
+
+        await enrollmentPage.navigate();
+        await enrollmentPage.assertPageHeading("Registrace");
+    });
+
+    test("Registration form should be visible, enabled and editable", {
+        annotation: {
+            type: "test description",
+            description: "Test který přejde na formulář registrace a zkontroluje, že se formulář správně zobrazil"
+        }
+    }, async ({ page }) => {
+        const enrollmentPage = new EnrollmentPage(page);
+
+        await enrollmentPage.assertFormElements();
+    });
+
+    test("New user should register", {
+        annotation: {
+            type: "test description",
+            description: "Test který provede validní registraci uživatele"
+        }
+    }, async ({ page }) => {
+        const enrollmentPage = new EnrollmentPage(page);
+        const { randomName, randomEmail, randomPassword } = await enrollmentPage.generateRandomUser();
+
+        await enrollmentPage.register(randomName, randomEmail, randomPassword);
+        await enrollmentPage.assertPageHeading("Přihlášky");
+        await enrollmentPage.assertUserIsLogged(randomName);
+    });
+
+    test("Existing user should not register", {
+        annotation: {
+            type: "test description",
+            description: "Test, který provede registraci uživatele s již existujícím emailem"
+        }
+    }, async ({ page }) => {
+        const enrollmentPage = new EnrollmentPage(page);
+
+        await enrollmentPage.register(userFullName, username, password);
+        await enrollmentPage.assertAlertMessage("Účet s tímto emailem již existuje");
+        await enrollmentPage.assertAlertCount(1);
+        await enrollmentPage.assertPageHeading("Registrace");
+    });
+
+    test("User with invalid password should not register", {
+        annotation: {
+            type: "test description",
+            description: "Test, který provede registraci uživatele s nevalidním heslem (obsahující pouze čísla)"
+        }
+    }, async ({ page }) => {
+        const enrollmentPage = new EnrollmentPage(page);
+        const { randomName, randomEmail } = await enrollmentPage.generateRandomUser();
```

Comment:
Líbí se mi jak si hraješ a zkoušíš! Super. Tady by mě napadla otázka, když ti test spadne, budeš vědět na jakém stringu/ uživateli spadla, když je pokaždé random? Je tvým cílem pokaždé spouštět test na jiných hodnotách?

Reply:
1) Debug - chybí mi zde chybové hlášky, které by znamenaly čitelnější chybu a snadnější debugging
2) Random data - za mě jsou dynamická data vždy lepší, ale máš pravdu, že to není úplně předmětem testu 🤔

Reply:
Chybové hlášky přidány.